Extension/src/LanguageServer/client.ts

```diff
@@ -134,7 +134,65 @@ function collectSettings(filter: (key: string, val: string, settings: vscode.Wor
         }
         if (filter(key, val, settings)) {
             previousCppSettings[key] = val;
-            result[key] = (key === "clang_format_path") ? "..." : String(previousCppSettings[key]);
+            switch (key.toLowerCase()) {
+                case "clang_format_path": {
+                    if (val) {
+                        switch (val.toLowerCase()) {
+                            case "/usr/bin/clang-format":
+                            case "/usr/local/bin/clang-format": {
+                                result[key] = String(previousCppSettings[key]);
+                                break;
+                            }
+                            default: {
+                                if (val.endsWith("clang-format.exe")) {
+                                    result[key] = "clang-format.exe";
+                                } else if (val.endsWith("clang-format")) {
+                                    result[key] = "clang-format";
+                                } else {
+                                    result[key] = "...";
+                                }
+                                break;
+                            }
+                        }
+                    } else {
+                        result[key] = "null";
+                    }
+                    break;
+                }
+                case "clang_format_style":
+                case "clang_format_fallbackstyle": {
+                    if (val) {
+                        switch (val.toLowerCase()) {
```

Comment:
same comment as above.  `val` must be confirmed to be a string before this can be called on it.